### DISCUSSION TITLE
Supply Status API not working

### DISCUSSION THREAD

**Discussion by zeroresearch** [gochain-io/docs#121] [Q&A]
v3.4.7 - Always returns an internal server error
config: 

```
_"alloc": {
    "0xF2cBFB1dd50601610e9c4a699780f14415e84e83": {
      "balance": "0x33b2e3c9fd0803ce8000000"_
```
Error:
```
_"code": -32000,
    "message": "unknown initial allocation"
  }
}_
```

**Comment by treeder** [comment]
Did you have all that configured when you first started it? If not, try deleting the data directory and start it again.

**Reply by zeroresearch** [reply]
Yes it was all configured when I first started it. 
Also I tried to set up a everything in a new environment and it didn't work.

**Comment by zeroresearch** [comment]
Is that it? No further ideas? 2 weeks nothing?

**Reply by treeder** [reply]
Can you give me some steps to reproduce?  What endpoint are you calling, etc?  I just started a new instance and not seeing any issues.

**Comment by zeroresearch** [comment]
Here’s what we did: 

1. Create an address for a node
    docker run --rm -v $PWD:/root gcr.io/gochain-core/gochain gochain --datadir /root/node --password /root/password.txt account new

2. Fill genesis.json
    Put address from p.1 to "signers", "voters","coinbase","alloc"
Set chain id to 9999 (own isolated network)
3. Fill .env file with network id and account

4. Generate genesis
docker run --rm -it -v $PWD:/gochain -w /gochain gcr.io/gochain-core/gochain gochain --datadir /gochain/node init genesis.json

5.Run the node
docker-compose up -d

6. Check balance of address from p.1
curl http://127.0.0.1:8545 --header "Content-Type: application/json" --request POST --data '{"jsonrpc":"2.0","id":0,"method":"eth_getBalance","params": ["0xfC734338B17dA90a220b0955eda3633E056Ad2B3", "latest"]}'
Reply:
 #{"jsonrpc":"2.0","id":0,"result":"0x33b2e52a232412860980000"}

7. Check allocation at genesis
curl http://127.0.0.1:8545 --header "Content-Type: application/json" --request POST --data '{"jsonrpc":"2.0","id":0,"method":"eth_genesisAlloc"}'
    
Reply:
{"jsonrpc":"2.0","id":0,"error":{"code":-32000,"message":"unknown initial allocation"}}

**Reply by zeroresearch** [reply]
[files2attach.zip](https://github.com/gochain/docs/files/7163434/files2attach.zip)
 These are the files I used

**Reply by treeder** [reply]
I think if the balance of the genesis address is ok (which it looks like it is) then you should be fine. I don't think that eth_genesisAlloc is even part of the [spec](https://eth.wiki/json-rpc/API) so I wouldn't use it or depend on it. 

Are you able to transfer from the coinbase account?

**Reply by zeroresearch** [reply]
Yes we can transfer from the coinbase account. But the supply status api give an error code on the explorer page because the rpc call gets the error code “unknown initial alloc” - the same call works for the official gochain roc so something must be wrong :(

**Reply by jmank88** [reply]
You will have to provide the custom genesis via a toml config file: https://github.com/gochain/gochain/blob/master/eth/config.go#L57
IIRC this was required because `init` doesn't write the initial state in a recoverable way.
The main and test nets 'just work' because their genesis data is bundled in from the source code.

**Reply by cppfuns** [reply]
> You will have to provide the custom genesis via a toml config file: https://github.com/gochain/gochain/blob/master/eth/config.go#L57
> IIRC this was required because `init` doesn't write the initial state in a recoverable way.
> The main and test nets 'just work' because their genesis data is bundled in from the source code.

Can you tell me specifically how to solve it? I built my own private network

**Reply by ghost** [reply]
Hi, Any update on this? how can it fetch the genesis data of the private network ?
